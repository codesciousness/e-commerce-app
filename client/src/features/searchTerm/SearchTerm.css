.SearchTerm {
    display: flex;
    background: white;
    border: 1px solid black;
    border-radius: 10px;
    width: 50%;
    margin: 2rem auto;
}

.SearchTerm__input {
    width: 100%;
    background: none;
    border-radius: 10px;
    padding: 0.75rem 2rem;
    outline: none;
}

.SearchTerm__icon {
    padding: 0.75rem 1rem;
}

/* Media Queries */

@media only screen and (max-width: 1000px) {

    .SearchTerm {
        width: 80%;
    }
}