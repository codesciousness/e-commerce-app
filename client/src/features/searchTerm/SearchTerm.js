import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import './SearchTerm.css';
import { setSearchTerm, clearSearchTerm, selectSearchTerm } from './searchTermSlice';

const SearchTerm = () => {
    const searchTerm = useSelector(selectSearchTerm);
    const dispatch = useDispatch();

    const handleChange = ({ target }) => {
        dispatch(setSearchTerm(target.value));
    };

    const handleClick = () => {
        dispatch(clearSearchTerm());
    };

    return (
        <form className="SearchTerm">
            <i className="fas fa-search fa-lg SearchTerm__icon"></i>
            <input className="SearchTerm__input" placeholder="Search..." autoFocus="autoFocus" onChange={handleChange} value={searchTerm} />
            {searchTerm && <i className="fas fa-times fa-lg SearchTerm__icon" onClick={handleClick} role="button"></i>}
        </form>
    );
}

export default SearchTerm;