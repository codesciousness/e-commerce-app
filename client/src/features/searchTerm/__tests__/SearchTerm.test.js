import React from 'react';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { create } from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import SearchTerm from '../SearchTerm';

const mockStore = configureStore([]);

describe('<SearchTerm />', () => {
    let store;

    beforeEach(() => {
        store = mockStore({
            searchTerm: ''
        });
    });

    test('SearchTerm component is defined', () => {
        expect(<SearchTerm />).toBeDefined();
    });
    test('matches the snapshot', () => {
        const component = create(
            <Provider store={store}>
                <SearchTerm />
            </Provider>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
    test('renders correctly', () => {
        const { container } = render(
            <Provider store={store}>
                <SearchTerm />
            </Provider>
        );
        expect(container.querySelector('.SearchTerm')).toBeInTheDocument();
        expect(screen.getByPlaceholderText(/Search/)).toBeInTheDocument();
    });
    test('dispatches correct action when input change occurs', () => {
        render(
            <Provider store={store}>
                <SearchTerm />
            </Provider>
        );
        const input = screen.getByPlaceholderText(/Search/);
        userEvent.type(input, 'c');
        const actions = store.getActions();
        const expectedPayload = {type: 'searchTerm/setSearchTerm', payload: 'c'};
        expect(actions).toEqual([expectedPayload]);
    });
    test('dispatches correct action when clear search button is clicked', () => {
        store = mockStore({
            searchTerm: 'c'
        });
        const { container } = render(
            <Provider store={store}>
                <SearchTerm />
            </Provider>
        );
        const button = container.querySelector('.fa-times');
        userEvent.click(button);
        const actions = store.getActions();
        const expectedAction = {type: 'searchTerm/clearSearchTerm'};
        expect(actions).toEqual([expectedAction]);
    });
    test('does not display clear search button when SearchTerm is falsy', () => {
        render(
            <Provider store={store}>
                <SearchTerm />
            </Provider>
        );
        let button = screen.queryByRole('button');
        expect(button).toBeNull();
        store = mockStore({
            searchTerm: 'c'
        });
        render(
            <Provider store={store}>
                <SearchTerm />
            </Provider>
        );
        button = screen.queryByRole('button');
        expect(button).toBeInTheDocument();
    });
});