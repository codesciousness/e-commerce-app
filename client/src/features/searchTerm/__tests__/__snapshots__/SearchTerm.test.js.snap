// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchTerm /> matches the snapshot 1`] = `
<form
  className="SearchTerm"
>
  <i
    className="fas fa-search fa-lg SearchTerm__icon"
  />
  <input
    className="SearchTerm__input"
    onChange={[Function]}
    placeholder="Search..."
    value=""
  />
  
</form>
`;
